### PR TITLE
Add nearest interpolation mode to volume rendering

### LINKED ISSUE #574: Bad 3D rendering of labels
## 🐛 Bug

I get some bad rendering of labels data in 3D, running the code in this gist:

https://gist.github.com/jni/587d8a946e5e86a8c0d3a7f8bee329cc

with `ipython --gui=qt` then `%run -i rendering.py`

Screen Shot 2019-10-08 at 12 45 45 pm

The issue seems to be that the layer values are interpolated, which causes the crazy fringing with the random colormap. Note that I played around with the interpolation method in the image layers, but the problem does persist after restarting Python.

Additionally, "mip" rendering is a poor choice for labels, because high-value labels appear on top of low-value ones, even when the low labels are in front of the high ones. However, translucent rendering did not fix the fringing problem, so it still looks terrible for me.

## To Reproduce

1. Download https://gist.github.com/jni/587d8a946e5e86a8c0d3a7f8bee329cc to a file named `rendering.py`
2. Launch `ipython --gui=qt` in that directory.
3. Run `%run -i rendering.py`

## Environment

 - napari Version (e.g., 1.0): master
 - OS (e.g., Linux): macOS 10.14.6
 - Python version: 3.7.3
 - Any other relevant information: vispy version 0.6.1

### PR DESCRIPTION
**Pull request by sofroniewn**
# Description
This PR follows on from #840 (which should be reviewed and merged first) and closes #574 by providing a `nearest` interpolation mode (in addition to default `linear` interpolation mode) for 3D volume rendering. Ultimately we may want to push these changes back to vispy, but for now as we are editing things frequently, I think it is fine to keep them here. Not that this improvement will be compatible with whatever colormap improvements come with #713 as it is only touching the interpolation code. Rendering of our image layers is also unchanged by this PR.

Here is what the labels layer now looks like:
better_3D_labels
which is much nicer!!

### CODE REVIEW

**Comment by sofroniewn** [comment]
I have made the corresponding PR here https://github.com/vispy/vispy/pull/1803

**Comment by sofroniewn** [comment]
@jni I am trying to make this PR only have the changes needed for the interpolation functionality, but instead the diff seems to have everything!! I tried a rebase with master after merging #840, but it seems to have just polluted the git history :-( Can you maybe help? The PR should only be adding about 20 lines in the Volume file instead of the whole thing ....

**Comment by jni** [comment]
@sofroniewn done! I'll merge once tests pass. For the record, what I did was:

- update my master
- fetch your branch
- `git rebase -i master` to do an interactive rebase
- delete the first two commits, which were redundant (they were the commits in #840)
- apply the rebase (which applied cleanly and only contained the small changes)
- force push to your branch.

Everything looks great now. I've renamed "i" to "interp" and you might want to do the same in your PR to VisPy, but other than that all good! :tada: This pair of PRs is just so fantastic for our 3D functionality!

**Review by jni — APPROVED** [review]

**Comment by royerloic** [comment]
This is great @sofroniewn !

**Pull request merged**
